packages/react-router/lib/dom/lib.tsx

```diff
@@ -94,7 +94,7 @@ import {
   useResolvedPath,
   useRouteId,
 } from "../hooks";
-import type { SerializeFrom } from "../types/route-data";
+import type { SerializeFrom } from "../types/internal-export/route-data";
```

Comment:
@pcattori It looks like we have some circular deps here which are probably fine but it might be something we look into cleaning up later?  `react-router/internal` imports `RouteFiles ` from `react-router` so it doesn't seem like `react-router` should reach back into `/internal` stuff?  

It looks like the underlying shared thing is `ServerDataFrom`/`ClientDataFrom` so maybe they should be exported as `UNSAFE_` from `react-router` and pulled into `/internal` that way alongside `RouteFiles`?